prov/util/src/util_mr_cache.c

```diff
@@ -399,6 +400,12 @@ struct ofi_mr_entry *ofi_mr_cache_find(struct ofi_mr_cache *cache,
 		goto unlock;
 	}
 
+	monitor = cache->monitors[entry->info.iface];
+	if (!monitor->valid(monitor, &entry->info, entry)) {
```

Comment:
Need to check for NULL monitor first like how `ofi_mr_cache_search` does.

Reply:
I believe if this returns NULL, this means an entry was added to the cache without a memory monitor entry which seems like a fatal flaw in the MR cache. I can add an assert. Thoughts?

Reply:
A 'passive' memory monitor will not verify that an entry is still valid until it is accessed.  So, the entry would have been valid when it was added, but the valid check now fails.  This is how both CUDA and Ze monitors work.  Only RoCR is active in this regard and intercepts GPU allocation/free calls to invalidate entries.